### PR TITLE
Split cd test module in implementation and api modules

### PR DESCRIPTION
**Pull request by tokle**

### CODE REVIEW

File: cloud-tenant-base/pom.xml [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+
+    <artifactId>cloud-tenant-base</artifactId>
+    <name>Vespa Cloud tenant base</name>
+    <version>7-SNAPSHOT</version>
+    <description>Parent POM for all Vespa Cloud applications.</description>
+    <url>https://github.com/vespa-engine</url>
+    <packaging>pom</packaging>
+
+    <parent>
+        <artifactId>tenant-base</artifactId>
```
Comment (bjorncs) [inline cloud-tenant-base/pom.xml:16]: Should this be `hosted-tenant-base`?

File: cloud-tenant-cd/pom.xml [hunk 2]
```diff
@@ -20,6 +18,11 @@
     </parent>
 
     <dependencies>
```
Comment (bjorncs) [inline cloud-tenant-cd/pom.xml:20]: We'll need to clean up these dependencies when we redesign this as a bundle

File: hosted-tenant-base/pom.xml [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+            <artifactId>container</artifactId>
+            <version>${vespaversion}</version>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>com.yahoo.vespa</groupId>
+            <artifactId>container-test</artifactId>
+            <version>${vespaversion}</version>
+            <scope>test</scope>
```
Comment (bjorncs) [inline hosted-tenant-base/pom.xml:85]: We'll have to override the scope to runtime for `cloud-tenant-base`, perhaps we need to move this definition down.

File: hosted-tenant-base/pom.xml [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.commons</groupId>
```
Comment (bjorncs) [inline hosted-tenant-base/pom.xml:88]: No idea why these exclusions are here.

File: hosted-tenant-base/pom.xml (hunk 3 above)
Comment (tokle) [reply hosted-tenant-base/pom.xml:85]: can't we just set the scope to runtime in hosted-tenant-base?

File: hosted-tenant-base/pom.xml (hunk 4 above)
Comment (tokle) [reply hosted-tenant-base/pom.xml:88]: 🤷

**Review by bjorncs — APPROVED** [review]

**Review by bjorncs — APPROVED** [review]

**Pull request merged**